File: cs/Flyweight.cs
using System.Collections;
using System;
namespace cs
{
    interface Flyweight
    {
        public void IncState(int value);
    }
    class ConcreteFlyweight : Flyweight
    {
        public int State = 0;

        public string Templite;

        public ConcreteFlyweight (string templite){
            this.Templite = templite;
        }
        public  void IncState(int value)
        {
            State += value;
        }

        public void Print (){
            Console.Write(this.Templite);
        }
    }
   
    class FlyweightFactory
    {
        Hashtable flyweights = new Hashtable();       
        public Flyweight GetFlyweight(string key)
        {
            if (!flyweights.ContainsKey(key))
                flyweights.Add(key, new ConcreteFlyweight(key));
            return flyweights[key] as Flyweight;
        }
    }
}

File: js/Flyweight.js
class Fliywight{
    constructor (){
        this.State = 0;
    }
    IncState(value){
        this.State += value
    }
}

class FlyweightFactory{
    constructor (){
        this.fliywights = {
        };
    }
    GetFlyweight(key){
        if (!this.fliywights[key])
            this.fliywights[key] = new Fliywight();
        return this.fliywights[key];    
    }
}

export {FlyweightFactory, Fliywight};